enhancements/hypershift/priority-class/priority-class.md

```diff
@@ -0,0 +1,275 @@
+- Override criteria:
+  - Adding the `hypershift.openshift.io/control-plane-priority-class` annotation in the HC spec will override any components with the `hypershift-control-plane` priority class to the value of the annotation.
+  - Adding the `hypershift.openshift.io/api-critical-priority-class` annotaation in the HC spec will override any components with the `hypershift-api-critical` priority class to the value of the annotation.
+  - Adding the `hypershift.openshift.io/etcd-priority-class` annotation in the HC spec will override any components with the `hypershift-etcd` priority class to the value of the annotation.
+  - The `hypershift-operator`, `system-cluster-critical`, and `system-node-critical` priorty classes will not be overriden by this enhancement. 
+- The `hostedcluster_controller` will reconcile the specified override priority classes from the HC annotations and mirror them for the HCP spec.
+- The CPO will reconcile the new override priority classes from the HCP spec, and propogate them to the various components determined by the override criteria.
+  - The `DeploymentConfig` struct will be used as it is currently to propogate changes in priority classes.
+
+### maxSurge
+At the moment, we are mitigating a Kubernetes rollout bug that can take down the service by setting `maxSurge` to `0`. Due to the bug, we experience an outage if a pod is restarted for any reason during a rollout, and since overriding priority classes initiates a rollout, it could potentially lead to an outage. 
+
+As a temporary workaround, we've adjusted `maxSurge` values for all HA control plane deployments to `0`.
+PR: https://github.com/openshift/hypershift/issues/1568
+
+We'll need to re-evaluate if this needs to be reverted if/when the Kubernetes bug is resolved. 
+Kubernetes issue: https://github.com/kubernetes/kubernetes/issues/108266
+
+### multus-admission-controller
+Introduced to the control plane in v4.12, the `multus-admission-controller` is a deployment that's within scope of this enhancement. Currently, the `multus-admission-controller` is managed by the `cluster-network-operator`, which also sets its priority class. Since `multus` isn't managed by HyperShift, it's not possible to override the priority class since CNO will squash and changes made.
+An issue has been opened against CNO to resolve this blocker: https://issues.redhat.com/browse/OCPBUGS-7942.
+
+
+### Phase 2 - Dynamic Adjustment of Priority Classes (WIP)
+Phase 2 of this enhancement will focus on bolstering Hypershift's capability by introducing dynamic adjustment of priority classes based on the workload's size. This will directly be tied to the HC's total node count.
+- Add new priority classes to the cmd/install assets for HyperShift Operator, and enable the utilization of these custom priority classes using a CLI flag. 
```

Comment:
>Add new priority classes to the cmd/install assets for HyperShift Operator,

Is this to let an admin of the HO to extend the ones defined in the proposal section above? Is this of ruse cases where the ones defined above are not flexible enough?